Classes/OldModel/Event.php

```diff
@@ -389,27 +389,51 @@ public function getPlacesWithCountry(): array
             return [];
         }
 
-        $countries = [];
-
-        $countryData = \Tx_Oelib_Db::selectMultiple(
-            'country',
-            'tx_seminars_sites LEFT JOIN ' .
-            'tx_seminars_seminars_place_mm ON tx_seminars_sites' .
-            '.uid = tx_seminars_seminars_place_mm.uid_foreign' .
-            ' LEFT JOIN tx_seminars_seminars ON ' .
-            'tx_seminars_seminars_place_mm.uid_local = ' .
-            'tx_seminars_seminars.uid',
-            'tx_seminars_seminars.uid = ' . $this->getUid() .
-            ' AND tx_seminars_sites.country <> ""' .
-            \Tx_Oelib_Db::enableFields('tx_seminars_sites'),
-            'country'
-        );
+        $queryBuilder = self::getQueryBuilderForTable('tx_seminars_sites');
+        $countryData = $queryBuilder
+            ->select('country')
+            ->from('tx_seminars_sites')
+            ->leftJoin(
+                'tx_seminars_sites',
+                'tx_seminars_seminars_place_mm',
+                'place_mm',
+                $queryBuilder->expr()->eq(
+                    'tx_seminars_sites.uid',
+                    $queryBuilder->quoteIdentifier('place_mm.uid_foreign')
+                )
+            )
+            ->leftJoin(
+                'place_mm',
+                'tx_seminars_seminars',
+                'seminars',
+                $queryBuilder->expr()->eq(
+                    'place_mm.uid_local',
+                    $queryBuilder->quoteIdentifier('seminars.uid')
+                )
+            )
+            ->where(
+                $queryBuilder->expr()->eq(
+                    'seminars.uid',
+                    $queryBuilder->createNamedParameter($this->getUid(), \PDO::PARAM_INT)
+                ),
+                $queryBuilder->expr()->neq(
+                    'tx_seminars_sites.country',
+                    $queryBuilder->createNamedParameter('', \PDO::PARAM_STR)
+                )
+            )
+            ->groupBy('country')
+            ->execute()
+            ->fetchAll();
 
-        foreach ($countryData as $country) {
-            $countries[] = $country['country'];
-        }
+        $countryData = array_filter($this->getPlacesAsArray(), function ($place) {
+            if (isset($place['country']) && !empty($place['country'])) {
```

Comment:
AFAICT,  `$place['country']` will always be set. So we can drop this check.

We can also make the anonymous function static and add parameter and return type declarations.